src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/entity/ActionPlanAppointmentMeetingMethod.kt

```diff
@@ -0,0 +1,36 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity
+
+import com.vladmihalcea.hibernate.type.basic.PostgreSQLEnumType
+import org.hibernate.annotations.Fetch
+import org.hibernate.annotations.FetchMode
+import org.hibernate.annotations.Type
+import org.hibernate.annotations.TypeDef
+import java.util.UUID
+import javax.persistence.Entity
+import javax.persistence.EnumType
+import javax.persistence.Enumerated
+import javax.persistence.Id
+import javax.persistence.ManyToOne
+import javax.persistence.Table
+import javax.validation.constraints.NotNull
+
+@Entity
+@Table(name = "action_plan_appointment_meeting_method")
+@TypeDef(name = "meeting_method_code", typeClass = PostgreSQLEnumType::class)
+data class ActionPlanAppointmentMeetingMethod(
+  @Id val id: UUID,
+  @Type(type = "meeting_method_code")
+  @Enumerated(EnumType.STRING)
+  @NotNull var meetingMethodCode: MeetingMethodCode,
+  var videoCallAddress: String? = null,
+  var npsOfficeCode: String? = null,
+  @ManyToOne @Fetch(FetchMode.JOIN)
+  var otherMeetingLocation: MeetingLocation? = null,
```

Comment:
addresses are notoriously difficult to store correctly when split into constituent parts (house name / number, street address, city, county, postal code, country etc etc).

my vote to keep this is simple as possible would be a single text field.

happy to be overruled on this though. 

cc @sldblog

Reply:
furthermore, if we used a plain text field for this, we could trim down this entity to simply be 

`{ id: UUID, type: MeetingMethodCode, details: String }`

no optional fields, and the details corresponds to the meeting method - i.e. it's a url for video calls, a phone number for phone calls, an address for in person meetings etc 

what do you think?